src/serac/physics/materials/solid_material.hpp

```diff
@@ -128,6 +128,48 @@ struct NeoHookean {
   double G;        ///< shear modulus
 };
 
+/**
+ * @brief Neo-Hookean material model
+ *
+ */
+struct NeoHookeanWithFieldDensity {
+  using State = Empty;  ///< this material has no internal variables
+
+  /**
+   * @brief stress calculation for a NeoHookean material model
+   *
+   * When applied to 2D displacement gradients, the stress is computed in plane strain,
+   * returning only the in-plane components.
+   * @return The first Piola stress
+   */
+  template <typename T, int dim, typename Density>
+  SERAC_HOST_DEVICE auto pkStress(State& /* state */, const tensor<T, dim, dim>& du_dX, const Density&) const
+  {
+    using std::log1p;
+    constexpr auto I = Identity<dim>();
+    auto lambda = K - (2.0 / 3.0) * G;
+    auto B_minus_I = dot(du_dX, transpose(du_dX)) + transpose(du_dX) + du_dX;
+
+    auto logJ = log1p(detApIm1(du_dX));
```

Comment:
```suggestion
    constexpr auto I = Identity<dim>();
    auto lambda = K - (2.0 / 3.0) * G;
    auto B_minus_I = dot(du_dX, transpose(du_dX)) + transpose(du_dX) + du_dX;

    auto logJ = std::log1p(detApIm1(du_dX));
```
Nitpick: personally adverse to `using` statements unless they make readability better due to being used multiple times.

Reply:
unfortunately, that 'using' is required to resolve the somewhat ambiguous scope of overloaded functions in serac.  Due to the templates, the types of the argument to log1p will be different in different contexts, so your change wont actually compile when gradients are being computed.  This is something I've wanted to fix for some time, but haven't had the bandwidth.  I think all math functions need to be in the serac:: namespace and reimplemented to take double as argument.  Then we replace all these calls with, eg. serac::log1p or serac::sqrt, etc.

Reply:
Ah, while unfortunate that makes sense. A comment would be helpful then something like "resolve ambigous log1p overload"

Reply:
Made an issue #1349